src/templates/plugin.main.js

```diff
@@ -153,14 +153,20 @@ export default async (context) => {
     let redirectPath = ''
 
     const isStaticGenerate = process.static && process.server
+    const isSameLocale = getLocaleFromRoute(route) !== newLocale
+    const isPrefixAndDefaultStrategy = options.strategy === Constants.STRATEGIES.PREFIX_AND_DEFAULT
+    const isDefaultLocale = newLocale === options.defaultLocale
+    const isEnabledRedirectForRoute = isEnabledRedirectForPath(route.fullPath, app.i18n.localeCodes, options.disableDefaultRedirect)
+
     // Decide whether we should redirect to a different route.
     if (
       !isStaticGenerate &&
       !app.i18n.differentDomains &&
       options.strategy !== Constants.STRATEGIES.NO_PREFIX &&
       // Skip if already on the new locale unless the strategy is "prefix_and_default" and this is the default
-      // locale, in which case we might still redirect as we prefer unprefixed route in this case.
-      (getLocaleFromRoute(route) !== newLocale || (options.strategy === Constants.STRATEGIES.PREFIX_AND_DEFAULT && newLocale === options.defaultLocale))
+      // locale, in which case we might still redirect as we prefer unprefixed route in this case, but let user a
+      // possibility to disable this behavior by switching disableDefaultRedirect option to true.
+      (isSameLocale || (isPrefixAndDefaultStrategy && isDefaultLocale && isEnabledRedirectForRoute))
```

Comment:
I feel like having this would only solve part of the issue as linking to such page using `localePath('about')` would still resolve and navigate to locale-less path. If you are willing to work on it more to also handle that case then I'll be willing to accept it.

Note that if this would be handled in the path resolving code (`localePath` and co.) then you probably wouldn't even need to change this part.